Mathlib/RingTheory/Nilpotent/GeometricallyReduced.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2025 Dion Leijnse. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dion Leijnse
+-/
+import Mathlib.Algebra.Lie.OfAssociative
+import Mathlib.FieldTheory.IsAlgClosed.AlgebraicClosure
+import Mathlib.LinearAlgebra.TensorProduct.Finiteness
+import Mathlib.RingTheory.Flat.FaithfullyFlat.Basic
+import Mathlib.RingTheory.Henselian
+
+/-!
+# Geometrically reduced algebras
+
+In this file we introduce geometrically reduced algebras.
+For a field `k`, we say that a `k`-algebra `A` is geometrically reduced (`IsGeometricallyReduced`)
+if the tensor product `A ⊗[k] AlgebraicClosure k` is reduced.
+
+## Main results
+- `FlatBaseChangeIsReduced.of_FG`: if `k` is a commutative ring and `A` and `C` are `k`-algebras,
+  such that `C ⊗[k] B` is reduced for all finitely generated `k` subalgebras `B` of `A`, then
+  `C ⊗[k] A` is reduced.
+
+## References
+- See [https://stacks.math.columbia.edu/tag/05DS] for some theory of geometrically reduced algebras.
+  Note that their definition differs from the one here, we still need a proof that these are
+  equivalent.
+
+-/
+
+open TensorProduct
+
+noncomputable section
+variable {k : Type} {A : Type} [Field k] [Ring A] [Algebra k A]
```

Comment:
Here, we would like to have statements which apply to algebra any universe, not just `Type 0`. With our tensor product API, I believe that there should not be issues if we do `k : Type u` and `A : Type v`.